superset/views/core.py

```diff
@@ -991,7 +991,9 @@ def get_form_data(self, slice_id=None):
         slice_id = form_data.get('slice_id') or slice_id
         slc = None
 
-        if slice_id:
+        # Include the slice_form_data if form_data contains no datasource
+        datasource_info = form_data.get('datasource') or datasource_id
+        if slice_id and (use_slice_data or not datasource_info):
```

Comment:
`datasource_info` is the name of a local method and thus you probably should use a different name. Does `form_data.get('datasource')` contain the datasource ID?